rero_ils/modules/documents/dojson/contrib/marc21tojson/utils.py

```diff
@@ -1743,44 +1726,89 @@ def do_work_access_point(marc21, key, value):
 
     if not work_access_point.get('title'):
         error_print('WARNING WORK ACCESS POINT:', marc21.bib_id,
                     marc21.rero_id, 'no title')
         return None
-    agent = work_access_point.get('agent', {})
-    if agent and not agent.get('preferred_name'):
+    agent = work_access_point.get('agent', {}).get('entity', {})
+    if agent and not agent.get('authorized_access_point'):
         work_access_point.pop('agent')
-    return work_access_point or None
+    return _format_work_access_point(work_access_point) or None
 
 
 def do_work_access_point_240(marc21, key, value):
     """Get work access point from 240."""
-    work_access_points = {}
+    work_access_point = {}
     part_list = TitlePartList(
         part_number_code='n',
         part_name_code='p'
     )
     part_selection = {'n', 'p'}
     for blob_key, blob_value in get_field_items(value):
-        if blob_key in {'a'}:
-            title = remove_trailing_punctuation(
+        if blob_key == 'a':
+            work_access_point['title'] = remove_trailing_punctuation(
                 blob_value.replace('\u009c', ''))
-            work_access_points['title'] = title
-
         if blob_key in part_selection:
             part_list.update_part(blob_value, blob_key, blob_value)
 
     if field_100 := marc21.get_fields('100'):
         if agent := build_agent(marc21, '100', field_100[0]['subfields'])[1]:
-            work_access_points['agent'] = agent
+            work_access_point['agent'] = {
+                'entity': {
+                    'type': agent.get('type'),
+                    'authorized_access_point':
+                        create_authorized_access_point(agent)
+                }
+            }
 
     if the_part_list := part_list.get_part_list():
-        work_access_points['part'] = the_part_list
+        work_access_point['part'] = the_part_list
+
+    if work_access_point:
+        return _format_work_access_point(work_access_point)
+
+
+def _format_work_access_point(access_point):
+    """Format a structured work_access_points as a simple auth_access_point.
 
-    if work_access_points:
-        return work_access_points
+    During transformation process of original data, we build a list of
+    structured work_access_point ; we need to flat this structure on a simple
+    'authorized_access_point' string accepted into the document JSONSchema.
+
+    :param access_point: a structured work_access_point.
+    :returns the flatten authorized_access_points for this work.
+    """
+    label = ''
+    agent_entity = access_point.get('agent', {}).get('entity', {})
+    if agent_label := agent_entity.get('authorized_access_point'):
+        label = f'{agent_label}. '
+    label += f"{access_point['title']}. "
+    for part in access_point.get('part', []):
+        for key in ['partNumber', 'partName']:
+            if key in part:
+                label += f"{part[key]}. "
+    if misc_info := access_point.get('miscellaneous_information'):
```

Comment:
This can be probably done using a list: `for field in ['miscellaneous_information', 'language', ...]:`